core/src/test/java/org/elasticsearch/rest/action/RestActionsTests.java

```diff
@@ -66,6 +66,14 @@ public void testParseTopLevelBuilderEmptyObject() throws IOException {
         }
     }
 
+    public void testParseTopLevelBuilderMalformedJson() throws IOException {
+        String requestBody = "\"\"";
```

Comment:
Can you add a test for the cases `"\"someString\""` and maybe also make sure that we don't barf for the empty String, but simply return null like in the case of "{}"?

Reply:
Extended tests to cover this :)